### PR TITLE
Add __str__ and nbytes.

### PR DESCRIPTION
**Pull request by rhattersley**

### CODE REVIEW

**Comment by bjlittle** [comment]
@rhattersley pep8 :stuck_out_tongue_winking_eye:

**Comment by rhattersley** [comment]
:unamused:

**Comment by rhattersley** [comment]
Fixed. :expressionless:

**Comment by bjlittle** [comment]
@rhattersley Nice :+1:

File: biggus/__init__.py [hunk 1]
```diff
@@ -485,6 +485,21 @@ class Array(object):
...
+        if nbytes < (1 << 10):
+            size = '{} B'.format(nbytes)
+        elif nbytes < (1 << 20):
+            size = '{:.02f} KiB'.format(nbytes / (1 << 10))
+        elif nbytes < (1 << 30):
+            size = '{:.02f} MiB'.format(nbytes / (1 << 20))
+        elif nbytes < (1 << 40):
+            size = '{:.02f} GiB'.format(nbytes / (1 << 30))
+        else:
+            size = '{:.02f} TiB'.format(nbytes / (1 << 40))
```
Comment (pelson) [inline biggus/__init__.py]: This ``size`` is more useful to me as a human than the nbytes property. Would you be happy to extract this functionality into a function which can be passed any array (biggus or any other type which has a nbytes attribute)?

File: biggus/__init__.py [hunk 2]
```diff
@@ -485,6 +485,21 @@ class Array(object):
 
     __hash__ = None
 
+    def __str__(self):
+        nbytes = self.nbytes
+        if nbytes < (1 << 10):
+            size = '{} B'.format(nbytes)
+        elif nbytes < (1 << 20):
+            size = '{:.02f} KiB'.format(nbytes / (1 << 10))
```
Comment (QuLogic) [inline biggus/__init__.py]: This would be the same as `nbytes >> 10`, wouldn't it?

Comment (rhattersley) [inline biggus/__init__.py]: They're not quite the same. `nbytes / 1024` gives a floating point result, `nbytes >> 10` gives an integer (equivalent to `nbytes // 1024`).

File: biggus/__init__.py (hunk 1 above)
Comment (rhattersley) [inline biggus/__init__.py]: Sure.

Comment (rhattersley) [inline biggus/__init__.py]: See #128.

**Pull request merged**